components/feature/push/src/main/java/mozilla/components/feature/push/Connection.kt

```diff
@@ -154,6 +159,14 @@ internal class RustPushConnection(
         return pushApi.unsubscribeAll()
     }
 
+    @GuardedBy("this")
+    override suspend fun containsSubscription(scope: PushScope): Boolean = synchronized(this) {
+        val pushApi = api
```

Comment:
Why do you need a local variable? you already `synchronzied` the call to this, so as long as every access to `api` is also synchronzied it seems unnecessary?